### ISSUE TITLE
Altered Files that I can not detect, Permissions, sticky bits etc. All good , file/folder during Samba share messed up

### ISSUE THREAD

**Issue by SaskOldFart** [linuxmint/nemo#2653]
```
 * 4.8.4
 * Is issue with desktop or windowed nemo?
 * Mint 20+20.1
 * Graphics hardware *and* driver used
 * 64 bit
 ```
**Issue**
I have 5 Hard disks, Each own thing (downloads)(music)etc.  I download everything to 1 drive and move after checked to other drive and other folders using NEMO.   Now in this server/workstation everything appears fine.   When shared through Samba it either disappears or inaccessible.  So I verified on windows 10 + another Linux mint workstation.   Also, All copy/move operations have been done on the server itself!!   Linux mint / Nemo

So to be clear , I have shared folders on other drives which work.
I copy or move using nemo and sometimes this issue occurs.

So again on any workstation the files / folders are missing or not able to access.

Now I verified all permissions and even checked sticky bits etc. tried resetting permissions and no success, they appear identical!!

**Steps to reproduce**
This only started in Linux Mint 20
I have a Linux machine just for downloading.  I move all files to my main server.
So then I can copy or move a file to ANOTHER hard disk and this is when it happens
Permissions are identical to working files. Including Owners!!
I can access and do what ever I want at the server itself.
Only seems to be an issue on Shared Folders for Samba
Once file or folder seems to be copied it is somehow changed?  It never works Properly.

If I copy the affected files within Nemo - No success. THOUGH cp or krusader work fine..  In fact I can take the files affected and just make copies and they are fine, This only happens with NEMO

**Expected behavior**

**Other information**

Work-Around
1)I can use terminal cp commands to copy everything to another hard disk and then nemo works fine
2) I can copy files within the same directory with cp command in terminal and copies are fine, originals must be deleted
3)I tried Krusader and works fine with everything, in fact using it to fix all of my files
I also checked advanced permissions & everything I can think of and everything matches!!  

Mint was a clean install to 20.0 <- had issue and then 20.1 same

I am somewhat new to linux and it just seems like there is a hidden feature or something and its tagging certain files.  It's very strange.  For now I must quit NEMO and I love using it...

**Comment by Jeremy7701** [comment]
"2) I can copy files within the **same directory** with cp command in terminal and copies are fine, originals must be deleted..."
This would imply that the `cp` command does something that fixes the problem.
I'd suggest obtaining UID/GID information on a before/after basis by comparing the output of `stat filename`.

**Comment by SaskOldFart** [comment]
ok.  so here is an example...   
I have a directory which I can see and everything is fine from the server.   Samba can see but in windows 10 shows a little x in bottom left corner of icon.. 
I have a file 1.txt   in this directory which is messed up..
I can use cp 1.txt 2.txt
now 2.txt shows up in samba and works on client in that directory

so here is stat of 1.txt

  File: 1.txt
  Size: 44        	Blocks: 8          IO Block: 4096   regular file
Device: 811h/2065d	Inode: 23199747    Links: 1
Access: (0777/-rwxrwxrwx)  Uid: ( 1000/ myname)   Gid: ( 1000/ myname)
Access: 2021-01-28 17:13:06.379088000 -0600
Modify: 2021-01-13 23:23:21.273037000 -0600
Change: 2021-01-29 14:06:17.698633951 -0600
 Birth: -

here is stat of 2.txt 

  File: 2.txt
  Size: 44        	Blocks: 8          IO Block: 4096   regular file
Device: 811h/2065d	Inode: 23199748    Links: 1
Access: (0775/-rwxrwxr-x)  Uid: ( 1000/ myname)   Gid: ( 1000/ myname)
Access: 2021-01-29 14:07:02.425695747 -0600
Modify: 2021-01-29 14:06:57.321802962 -0600
Change: 2021-01-29 14:06:57.321802962 -0600
 Birth: -


Now i have given 1.txt full rwx as well.   I am at a loss.....   Have tried stat before

As I said this issue only happens with NEMO copying from 1 hard disk to another, started mint 20 and doesn't do it all the time.

I have kept a few files to help get this bug resolved.

once again if I take 1.txt and use nemo in same directory to copy to 2.txt it will replicate this glitch.   cp command or Krusader FIX / Do not have problems..   Something is altered but I am at a loss.....   I have spent a week on this reinstalling Linux on multiple pc's and finally it's definitely NEMO

Willing to try anything to help figure this out..

**Comment by Jeremy7701** [comment]
1.txt `Access: (0777/-rwxrwxrwx)`
2.txt `Access: (0775/-rwxrwxr-x)`
So `1.txt` has rwx access for owner, group and everyone else, incidentally this is probably "too much access" !
`2.txt` has rwx access for owner & group but only read and execute access for everyone else.

Note that 2.txt has **less** permission than 1.txt - there is a good reason for this:-
Depending on a certain configuration setting, newly created files (or downloaded files) are given certain permissions; by default this is:-
owner: rw
group: rw
others: r
Note that **no** executable permissions are granted for files by default.
Check  your default configuration setting by entering the `umask` command (no parameters)
It should display **0002**
It is not recommended that this should be changed!

I'm mystified where all the extra exec and write permissions are coming from.
Is the Samba user configured to need write access to files it doesn't own?

**Comment by SaskOldFart** [comment]
1st - As stated above I gave the file full permissions for troubleshooting purposes and no mater what it doesn't affect it.  I've tried removing all access, putting it back.  I've tried special permissions all on , all off.  Thinking this might reset something?  NO affect

i tried your umask it did return 0002

Samba user is the same as the user on the server.   I do have rwx permissions on this particular shared drive and as stated everything works great as it should.   Other shares on the same server have same results, read only etc show same results.

This truly is just something I have found NEMO do and as I said, I can take the affected files , just make copies in the same location and they work,(MUST BE DONE AT SERVER THOUGH) I then delete affected files, rename to original and I'm in business.

This just blows my mind as it affect the file somehow that I can not determine.  As I said before once i notice a file acting up nemo keeps replicating the affected files.   cp command or Krusader both on copy actually have no issue or in fact repair it..   Using as the same user no SUDO.   Same users...

***
so now something else,  I made a copy of affected file in NEMO which is still affected.
okay so this is interesting,  I took a copy of affect file and called it test.txt
I then used the mv command and did mv test.txt 3.txt  and guess what the problem still exists.
renaming in Krusader had no affect either.

so something is  with this exact file..

**Comment by Jeremy7701** [comment]
`mv` [within a filesystem] doesn't move the data - it just changes the name in the directory.
The problem almost certainly lies with your Samba configuration.
You shouldn't be getting all those 'x' permissions unless you are setting them somewhere.

**Comment by SaskOldFart** [comment]
I disagree with Samba because of stated below. Also I am doing the file moving on the server itself.

I am just tinkering with the permissions on the server to test the files.  So somewhat ignore if its 755 or 777.   I'm just trying to give max permissions to eliminate those issues.
You guys just have to trust me, its not from copying in Samba.  
Yes samba is picking up this issue but why? I am literally just trying to read files from a share.. 
All of the work is done on the server..   So how in the world are some files affected?  Also its ONLY NEMO that causes this.
It's a bug in NEMO and other people will pick it up eventually.  I'm telling you this same exact setup works fine in mint 19.3
I have also checked advanced permissions (set userid setgroup id sticky) EVERYTHING MATCHES!!  files exact..

**Steps to reproduce**

So i watch a share from a client linux /win10 (changes show up immediately)
I go to the server and do what I said above.  I can copy a file with cp to same folder like cp 1.txt 2.txt ->Bang file works perfect in clients.... Shows up and can view right now!

*This is Important*
copy paste with new name, same directory
If I copy the file in NEMO it makes another affected file.. No access to clients, only can see the name.

How can you explain this?  
Also Krusader works fine, its a file manager copying the exact same file.
cp & krusader will fix the files and NEMO replicates the issue?
Yes Samba is picking it up but why?  Something is changing which is beyond me but someone may have an idea.

I am no genius with this but there is a glitch..  And somewhere somehow the file is affected.  I might have to make a video. or do screenshots

**Comment by Jeremy7701** [comment]
OK I give up, someone else has to help you.
But first check that only a single user ever manipulates those files (and it isn't root).
"Advanced permissions" are definitely nothing to do with it though.
